sirepo/pkcli/jupyterhublogin.py

```diff
@@ -58,9 +60,21 @@ def maybe_create_sirepo_user(qcall, email, display_name):
         with qcall.auth.logged_in_user_set(u, method=sirepo.auth.METHOD_EMAIL):
             n = sirepo.sim_api.jupyterhublogin.create_user(
                 qcall=qcall,
-                check_dir=True,
             )
             qcall.auth_db.model("UserRole").add_roles(
                 roles=[sirepo.auth_role.for_sim_type("jupyterhublogin")],
             )
         return PKDict(email=email, jupyterhub_user_name=n)
+
+
+def old_user_dirs():
+    """Outputs commands to delete dirs for Jupyter users that are not in database."""
+    with sirepo.quest.start() as qcall:
+        m = ""
+        r = qcall.auth_db.model("JupyterhubUser").search_all_for_column("user_name")
+        for d in pkio.sorted_glob(
+            sirepo.sim_api.jupyterhublogin.cfg().user_db_root_d.join("*")
+        ):
+            if not os.path.basename(d) in r:
```

Comment:
> We have some directories with begin with a capital letter that are not users, e.g. USPAS-Instructor. These directories should not be deleted. You'll need to exclude those.

Couldn't valid usernames also begin with a capital letter? I think I either need a complete list to exclude here, or whoever is deleting could check against a list before deleting anything. I took @e-carlin's suggestion of removing the "rm -rf" and just outputting the dir.


> There was (is?) a bug where the user name contains _5F. These should be called out and fixed. This could be an echo in the output pointing out these names.

Can you please provide more context? What bug would it cause here? Maybe this should be its own pkcli to identify these problematic usernames?

Reply:
> Couldn't valid usernames also begin with a capital letter?

No. However, this is not clear, and there are many defects. api_authEmailLogin [enforces lower case](https://github.com/radiasoft/sirepo/blob/1789d23f51556178466ae83d7dc9e7cb323f80e3/sirepo/auth/email.py#L121), but this is definitely insufficient. `create_user_from_email` should enforce lower case, or better, `auth_db.email.__init__` should override super and force unverified_email and user_name in `fields` to be lower case.  

pkcli.admin and jupyterhublogin should force the emaila argument to lower case.

Reply:
> Can you please provide more context? What bug would it cause here? Maybe this should be its own pkcli to identify these problematic usernames?

I think as long as we are cleaning up invalid/old names. We should do that, since that's the name of the function.

#3748 describes the problem. 5F is _ (underscore) in ascii. There was a double encoding going on. I think we fixed it, but it may still be there. We restrict the jupyter users to a specific syntax. I think I just figured out the bug. I'll document in #3748.

Reply:
[5fcc5e6](https://github.com/radiasoft/sirepo/pull/7027/commits/5fcc5e68653ffaa21bfd7912d81f95c61e304a11)
[eae3cde](https://github.com/radiasoft/sirepo/pull/7027/commits/eae3cdeac182f2aa627c0d92b2abe06a9ce5d86d)